ArjunaJTA/jta/tests/classes/com/hp/mwtests/ts/jta/recovery/nonuniquexids/ImportNonUniqueBranchTest.java

```diff
@@ -0,0 +1,116 @@
+import com.arjuna.ats.arjuna.recovery.RecoveryModule;
+import com.arjuna.ats.internal.arjuna.recovery.PeriodicRecovery;
+import com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule;
+import com.arjuna.ats.internal.jta.transaction.arjunacore.jca.SubordinationManager;
+import com.arjuna.ats.jta.common.JTAEnvironmentBean;
+import com.arjuna.ats.jta.xa.XATxConverter;
+import com.arjuna.common.internal.util.propertyservice.BeanPopulator;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import javax.resource.spi.XATerminator;
+import javax.transaction.Transaction;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+import java.util.ArrayList;
+import java.util.List;
+
+// unit test based on a Jonathan Haliday's code
+public class ImportNonUniqueBranchTest {
+    private static XARecoveryModule xaRecoveryModule;
+    private static PeriodicRecovery periodicRecovery;
+
+    @BeforeClass
+    public static void beforeClass() {
+        RecoveryEnvironmentBean environmentBean = BeanPopulator.getDefaultInstance(RecoveryEnvironmentBean.class);
+        List<String> moduleNames = new ArrayList<>();
+        moduleNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.SubordinateAtomicActionRecoveryModule");
+        moduleNames.add("com.arjuna.ats.internal.jta.recovery.arjunacore.XARecoveryModule");
+        environmentBean.setRecoveryModuleClassNames(moduleNames);
+        environmentBean.setRecoveryBackoffPeriod(1);
+
+        BeanPopulator.getDefaultInstance(JTAEnvironmentBean.class).setXAResourceRecordWrappingPlugin(new XAResourceRecordWrappingPluginImpl());
+
+        periodicRecovery = new PeriodicRecovery(false, false);
+
+        for (RecoveryModule recoveryModule : periodicRecovery.getModules()) {
+            if (recoveryModule instanceof XARecoveryModule) {
+                xaRecoveryModule = (XARecoveryModule)recoveryModule;
+                break;
+            }
+        }
+    }
+
+    @AfterClass
+    public static void afterClass() {
+        periodicRecovery.shutdown(false);
+    }
+
+    @Test
+    public void testWrapped() throws Exception {
+        test(true);
+        Assert.assertEquals("resource commit error", 0, XAResourceImpl.getErrorCount());
+    }
+
+    @Test
+    public void testNotWrapped() throws Exception {
+        test(false);
+        Assert.assertNotEquals("resource commit should have failed", 0, XAResourceImpl.getErrorCount());
+    }
+
+    public void test(boolean wrap) throws Exception {
+        XAResourceImpl.clearErrorCount();
+        ResourceManager resourceManagerA = new ResourceManager("jndi:/A", wrap);
+        ResourceManager resourceManagerB = new ResourceManager("jndi:/B", wrap);
+
+        xaRecoveryModule.addXAResourceRecoveryHelper(resourceManagerA);
+        xaRecoveryModule.addXAResourceRecoveryHelper(resourceManagerB);
+
+        Xid xid = XATxConverter.getXid(new Uid(), false, 131080);
```

Comment:
Maybe use FormatConstants instead of the magic number, and/or comment to make clear the intent here is to force something other than JTA_FORMAT_ID so that createXid takes the right path.